DigitalLearningSolutions.Web/Helpers/RegistrationEmailValidator.cs

```diff
@@ -0,0 +1,78 @@
+﻿namespace DigitalLearningSolutions.Web.Helpers
+{
+    using DigitalLearningSolutions.Data.Services;
+    using DigitalLearningSolutions.Web.Extensions;
+    using DigitalLearningSolutions.Web.ViewModels.Register;
+    using Microsoft.AspNetCore.Mvc.ModelBinding;
+
+    public static class RegistrationEmailValidator
+    {
+        private const string WrongEmailForCentreErrorMessage =
+            "This email address does not match the one held by the centre";
+
+        private const string EmailInUseErrorMessage =
+            "A user with this email address is already registered; if this is you, " +
+            "please log in at this centre via the My Account page";
+
+        public static void ValidateEmailAddresses(
+            PersonalInformationViewModel model,
+            ModelStateDictionary modelState,
+            IUserService userService,
+            ICentresService? centresService = null
+        )
+        {
+            if (!model.Centre.HasValue)

```

Comment:
I've moved the centre null check to the RegisterAdminController, since we only need it for the register admin journey, not the register delegate one. We need to know that the centreId is not null in that journey, because if it is, then calling centresService.DoesEmailMatchCentre in this file will throw an error.